test-files/yaml_syntax.yaml

```diff
@@ -40,6 +40,28 @@ registers:
         type: u16
         start: 0
         end: 16
+  foo_0:
+    description: Instance of the Foo block
+    address: 64
+    block: foo
+    block_description: Doc comment for the Foo block
+    registers:
+      baudrate:
+        description: Baudrate register
+        address: 0
+        rw_type: rw
+        size_bits: 16
+        byte_order: le
+        fields:
+          value:
+            description: Baudrate value
+            type: u16
+            start: 0
+            end: 16
+  foo_1:
+    description: Second instance of the Foo block
+    address: 80
+    block: foo
```

Comment:
Hmmm... I'm gonna sleep on this. Not sure if I like one item to depend on another one

Reply:
So my guess is that the addresses of registers within a block are relative to the block?
So Baudrate here has an address of 64? And in foo_1 that same register has address 80?

Hmmm... Doesn't make it immediately obvious what the addresses are which could be annoying when reading, but also when writing the spec

Reply:
Yes, addresses are relative to the parent. You could have it use absolute addresses, but then you'd have to subtract out the parent address when instantiating a block at a different base address, which I think would be more confusing.

Reply:
Hmmm... There are pros and cons everywhere. I'm looking at what we can have right now without getting in the way and being extendible of the future.

What do you think about this?
```yaml
  foo_0:
    description: Instance of the Foo block
    base_address: 64 # I might make this optional in the future. Then the register addresses would be absolute instead of relative (so base_address would be 0)
    registers:
      baudrate:
        description: Baudrate register
        address: 0
        rw_type: rw
        size_bits: 16
        byte_order: le
        fields:
          value:
            description: Baudrate value
            type: u16
            start: 0
            end: 16
  foo_1:
    description: Second instance of the Foo block
    base_address: 80
    copy_of: foo_0
```

This would set up a nice couple of things in the future:
- I'd really like a `repeat` and `stride` option in the future and this can still be fitted in this model
- The `copy_of` sets precedent to add it to registers too. Also, this makes it seem more what it is. In your version you sort of define a block and create an instance of one in the same spot which looks weird to me. This `copy_of` feels more natural to me. With this you can just use the block type directly without generating it, only giving it a different address.
- We lose the block description, but I don't think that's too bad since the function to access the block is still documentable

Reply:
Oh and the reason why I want repeat blocks is because that unlocks APIs like `device.gpio(5).input().read()`.
This would allow selecting a gpio at runtime instead of at compile time